waspc/data/Generator/templates/server/src/auth/providers/oauth/createRouter.ts

```diff
@@ -43,47 +46,43 @@ export function createRouter(provider: ProviderConfig, initData: { passportStrat
           }
 
           // Wrap call to getUserFieldsFn so we can invoke only if needed.
-          const getUserFields = () => getUserFieldsFn(contextWithUserEntity, { profile: providerProfile });
+          const getUserFields = () => getUserFieldsFn ? getUserFieldsFn(contextWithUserEntity, { profile: providerProfile }) : Promise.resolve({});
           // TODO: In the future we could make this configurable, possibly associating an external account
           // with the currently logged in account, or by some DB lookup.
-          const user = await findOrCreateUserByExternalAuthAssociation(provider.id, providerProfile.id, getUserFields);
+          const auth = await findOrCreateAuthByAuthIdentity(provider.id, providerProfile.id, getUserFields);
 
-          const token = await sign(user.id);
+          const token = await sign(auth.{= userFieldOnAuthEntityName =}.id);
           res.json({ token });
       })
     )
 
     return router;
 }
 
-async function findOrCreateUserByExternalAuthAssociation(
-  provider: string,
-  providerId: string,
+async function findOrCreateAuthByAuthIdentity(
+  providerName: string,
+  providerUserId: string,
   getUserFields: () => ReturnType<GetUserFieldsFn>,
-): Promise<{= userEntityUpper =}> {
+) {
```

Comment:
Here I would also have return type.

Reply:
:rocket: ? It is much faster for me to just see the :rocket: then to go hunt down the change and check if it was done.